### PR TITLE
docs(material/tab-group): lacks documentation for mat-stretch-tabs

### LINKED ISSUE #19263: docs-bug(mat-tab-group): mat-tab-stretch exists and is not documented anywhere
#### Documentation Feedback

It would be very useful to have all properties that affect a given component clearly and visibly documented. Today I found out about mat-tab-stretch after hacking away for a while with no success. 

Scouring https://material.angular.io/components/tabs/api and related provided no hint of its existence, and was only mentioned on this S.O. thread - https://stackoverflow.com/questions/48839430/make-tab-labels-stretch-to-full-width-in-material-design-for-angular

**Affected documentation page:** https://material.angular.io/components/tabs/api

### PR DESCRIPTION
**Pull request by lukaonik**
adds input property to tab-group, so mat-stretch-tabs is exposed to public API

Fixes #19263

### CODE REVIEW

**Comment by lukaonik** [comment]
@googlebot I signed it!

**Comment by lukaonik** [comment]
> I'm not sure about adding an input only so something shows up in the docs. People might expect that if they set `mat-stretch-tabs="false"` the tabs won't stretch which isn't the case at the moment. We should either come up with an alternate way of exposing something to the docs that's not in TS, or make this into a proper input. WDYT @jelbourn?
yes, you're correct, what if we add it like `@HostBinding("attr.stretch-tabs") `property,

**Comment by jelbourn** [comment]
Yeah, we definitely don't want to add an `@Input` where it's not strictly necessary. I would just add a brief section about this to `tabs.md` (which is the Overview on the docs site)

**Comment by lukaonik** [comment]
what if we add
`@HostBinding("class.stretch")`
with appropriate boolean input property, and toggle class accordingly?

**Comment by jelbourn** [comment]
That would definitely be more consistent with other APIs, though we would have to go through a deprecation phase where both work

**Comment by lukaonik** [comment]
I think, it will work with previous API also, shall I try it?

**Comment by jelbourn** [comment]
Go ahead- once you make the change we can run a presubmit on Google's internal code and see if any issues come up

File: src/material/tabs/tab-group.spec.ts [hunk 1]
```diff
@@ -636,6 +637,29 @@ describe('MatTabGroup', () => {
...
+    beforeEach(() => {
+      fixture = TestBed.createComponent(StretchedTabGroup);
+      fixture.detectChanges();
+      element = fixture.debugElement.query(By.css('mat-tab-group')).nativeElement;
+    });
+
+    it('should add mat-stretch-tabs class when stretchTab is true', (() => {
+      fixture.componentInstance.stretchTab = true;
+      fixture.detectChanges();
+      expect(element.classList.contains('.mat-stretch-tabs')).toBe(false);
```
Comment (shametim) [inline src/material/tabs/tab-group.spec.ts:653]: Did you mean toBe(true)?

Comment (lukaonik) [reply src/material/tabs/tab-group.spec.ts]: yes definitely, nice catch thx :)

**Comment by jelbourn** [comment]
@andrewseguin PTAL

**Comment by DaSchTour** [comment]
As I just came across this issue and also the discussion about adding this as an input.
I would vote for adding this as an input and changing the styling to use a class instead of the attribute styling. As for me IntelliJ is complaining that the attribute mat-stretch-tabs is not allowed.

File: .vscode/.editorconfig [hunk 2]
```diff
@@ -0,0 +1,16 @@
+root = true
```
Comment (andrewseguin) [inline .vscode/.editorconfig:1]: This should be left out of the commit

File: src/material/tabs/tab-group.ts [hunk 3]
```diff
@@ -106,6 +107,13 @@ export abstract class _MatTabGroupBase extends _MatTabGroupMixinBase implements
   /** Subscription to changes in the tab labels. */
   private _tabLabelSubscription = Subscription.EMPTY;
 
+   /** Whether the tab group should grow to the size of the active tab. */
```
Comment (andrewseguin) [inline src/material/tabs/tab-group.ts:110]: Fix documentation statement

File: src/dev-app/mdc-tabs/mdc-tabs-demo.html [hunk 4]
```diff
@@ -26,8 +26,8 @@ <h2>Themed tabs</h2>
     <mat-tab label="Fourth" disabled>Content 4</mat-tab>
   </mat-tab-group>
 
-  <h2>Stretched tabs</h2>
-  <mat-tab-group mat-stretch-tabs>
+  <h2>Stretched tabs ( 'mat-stretch-tabs' is deprecated, use stretchTabs property )</h2>
```
Comment (andrewseguin) [inline src/dev-app/mdc-tabs/mdc-tabs-demo.html:29]: I don't think the extra comment is needed here

**Comment by zarend** [comment]
Closing because there is already an example using mat-tab-stretch.

**Pull request closed without merging**